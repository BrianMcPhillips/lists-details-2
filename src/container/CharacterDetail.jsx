import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import CharacterDetail from '../components/Characters/CharacterDetail';
import { getSingleCharacter } from '../services/rickAndMorty';

const CharacterDetailPage = () => {
  const [character, setCharacter] = useState([]);

  const { id } = useParams();
  useEffect(() => {
    getSingleCharacter(id)
      .then(character => setCharacter(character));
  }, []);
  return (
    <div data-testid="detail">
      <CharacterDetail {...character} />
    </div>
  );
};

export default CharacterDetailPage;
