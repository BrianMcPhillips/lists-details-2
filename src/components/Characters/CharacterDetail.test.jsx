import React from 'react';
import { render } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('Character component', () => {
  it('should render a Character to the page', () => {
    const { asFragment } = render(<CharacterDetail 
      name="Test McTest"
      status="mehhh"
      species="maybe human"
      image="http://www.placecage.com/200/200"
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
