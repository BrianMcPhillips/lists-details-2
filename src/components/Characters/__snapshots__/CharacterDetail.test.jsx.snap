// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component should render a Character to the page 1`] = `
<DocumentFragment>
  <h2>
    Test McTest
  </h2>
  <h4>
    mehhh
  </h4>
  <h3>
    maybe human
  </h3>
  <img
    alt="Test McTest"
    src="http://www.placecage.com/200/200"
  />
</DocumentFragment>
`;
